src/transformers/trainer.py

```diff
@@ -1616,7 +1616,7 @@ def train(
             if resume_from_checkpoint is None:
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
 
-        if resume_from_checkpoint is not None and not is_sagemaker_mp_enabled():
+        if resume_from_checkpoint is not None and not is_sagemaker_mp_enabled() and args.deepspeed is None:
```

Comment:
It's already skipped inside here:

https://github.com/huggingface/transformers/blob/619d51e01f326d298687912d1f65f8d460f2a6e2/src/transformers/trainer.py#L2090-L2092

Does it fail for you in some preliminary checks that happen before the code I quoted above?

Reply:
Perhaps let's change line 1619, to just:
```
if resume_from_checkpoint is not None:
```
and start `self._load_from_checkpoint` with:

```
    def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
        if is_sagemaker_mp_enabled() or args.deepspeed:
            # sagemaker gets resumes in ...
            # deepspeed gets resumes in deepspeed_init
            return
```
and remove the code I quoted in the previous comment?

Reply:
or keep the version you proposed and then delete the code I quoted 2 comments up.

Let's ask @sgugger since he would be the final judge anyway. Sylvain is it better to quickly exit from the function as I proposed or not enter it like Moshe proposed?

In either way we should make things consistent as they aren't now. (sagemaker does the 2nd way)

Reply:
@stas00 I personally think that simply skipping the _load_from_checkpoint method if deepspeed is used, is the most straightforward way to fix this issue.

Its only a single if statement, and deepspeed_init already takes care of resuming the checkpoint correctly, the only thing that the train() method needs to do, is to let deepspeed do its own thing.

As for consistency, each method (regular, sagemaker, deepspeed) resumes the checkpoint using different logic and separate functions, thus changing that could potentially introduce more bugs into the process.

Reply:
As for the error itself, it is raised since deepspeed checkpoints do not contain WEIGHTS_NAME = "pytorch_model.bin":

https://github.com/huggingface/transformers/blob/aa3787c8f0b1ee4d4fa74fe890a39cfeb601d5e5/src/transformers/trainer.py#L2073

Reply:
yes, that's what I was suspecting - I wonder why nobody reported it since ever. Thank you for that!

--------------------

ok, if you prefer it this way (and let's assume Sylvain would agree) in that case then please adapt your PR to remove this code 

https://github.com/huggingface/transformers/blob/619d51e01f326d298687912d1f65f8d460f2a6e2/src/transformers/trainer.py#L2090-L2092

and replacing the next line's `elif` with `if`

Reply:
and please see the next comment about CircleCI - apparently we have no control over it - so you have to go to CircleCI - logout, login and that might do the trick. I have no idea why your CircleCI creds have anything to do with our project. Sounds like a bug in CircleCI. Thanks.

Reply:
The CI is triggered and green. But @stas00 above suggestion (about removing a block) is not applied yet. I will let @mosheber and @stas00 to continue, and jump in  in case CI has further issue.